### PR TITLE
Add grpc connect backoff policy and reduce the dial timeout

### PR DESCRIPTION
**Pull request by sunby**
issue: #13256
Signed-off-by: sunby <bingyi.sun@zilliz.com>

### CODE REVIEW

File: internal/util/grpcclient/client.go [hunk 1]
```diff
@@ -11,14 +11,20 @@ import (
...
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/keepalive"
 )
 
+const (
+	dialTimeout      = 5 * time.Second
+	keepAliveTime    = 10 * time.Second
```
Comment (jaime0815) [inline internal/util/grpcclient/client.go:24]: can we support configuring these variables?

File: internal/util/grpcclient/client.go [hunk 2]
```diff
@@ -106,61 +112,62 @@ func (c *ClientBase) resetConnection(client interface{}) {
...
+			)),
+		grpc.WithStreamInterceptor(
+			grpcmiddleware.ChainStreamClient(
+				grpcretry.StreamClientInterceptor(grpcretry.WithMax(3),
+					grpcretry.WithCodes(codes.Aborted, codes.Unavailable),
+				),
+				grpcopentracing.StreamClientInterceptor(opts...),
+			)),
+		grpc.WithKeepaliveParams(keepalive.ClientParameters{
+			Time:                keepAliveTime,    // send pings every 60 seconds if there is no activity
```
Comment (jaime0815) [inline internal/util/grpcclient/client.go:148]: make sure the comment is clear and accurate

File: internal/util/grpcclient/client.go [hunk 3]
```diff
@@ -106,61 +112,62 @@ func (c *ClientBase) resetConnection(client interface{}) {
...
+			Timeout:             keepAliveTimeout, // wait 6 second for ping ack before considering the connection dead
+			PermitWithoutStream: true,             // send pings even without active streams
+		}),
+		grpc.WithConnectParams(grpc.ConnectParams{
+			Backoff: backoff.Config{
+				BaseDelay:  100 * time.Millisecond,
+				Multiplier: 1.6,
+				Jitter:     0.2,
```
Comment (jaime0815) [inline internal/util/grpcclient/client.go:156]: add a debug log when creating a new connection?

File: internal/util/grpcclient/client.go (hunk 1 above)
Comment (sunby) [reply internal/util/grpcclient/client.go:24]: > can we support configuring these variables?

I think the timeout params do not need to be configuable. They should be set to constant variable.

File: internal/util/grpcclient/client.go (hunk 2 above)
Comment (sunby) [reply internal/util/grpcclient/client.go:148]: agreed

Comment (sunby) [reply internal/util/grpcclient/client.go:148]: I will change it in next pr and we can move forward.

File: internal/util/grpcclient/client.go (hunk 3 above)
Comment (sunby) [reply internal/util/grpcclient/client.go:156]: I will add a log in next pr

**Pull request merged**